apps/mobile/src/features/settings/theme-modal.tsx

```diff
@@ -1,21 +1,67 @@
 import { RefObject } from 'react';
 import { useSharedValue } from 'react-native-reanimated';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
 
 import { CLOSED_ANIMATED_SHARED_VALUE, Modal } from '@/components/bottom-sheet-modal';
+import { ModalHeader } from '@/components/headers/modal-header';
+import { useToastContext } from '@/components/toast/toast-context';
+import { ThemeStore, useSettings } from '@/state/settings/settings.slice';
 import { BottomSheetModal } from '@gorhom/bottom-sheet';
 import { t } from '@lingui/macro';
+import { useTheme } from '@shopify/restyle';
 
-import { Text } from '@leather.io/ui/native';
+import { Box, Cell, SunIcon, Theme } from '@leather.io/ui/native';
 
 interface ThemeModalProps {
   modalRef: RefObject<BottomSheetModal>;
 }
 export function ThemeModal({ modalRef }: ThemeModalProps) {
   const animatedIndex = useSharedValue<number>(CLOSED_ANIMATED_SHARED_VALUE);
+  const { bottom } = useSafeAreaInsets();
+  const theme = useTheme<Theme>();
+  const settings = useSettings();
+  const { displayToast } = useToastContext();
```

Comment:
This component could be better broken apart imo

Ideal outer component API might be something like

```
export function ThemeModal({ modalRef }: ThemeModalProps) {
  return (
    <ThemeModalLayout>
      <ThemeSwitcher activeTheme=}{ onUpdateTheme={() => {}}  />
    </ThemeModalLayout>
  );
}
```